src/components/multi-action-button/multi-action-button.test.tsx

```diff
@@ -0,0 +1,157 @@
+
+test("should render with provided 'text'", () => {
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toBeVisible();
+});
+
+test("should render with provided 'subtext' when 'size' is 'large'", () => {
+  render(
+    <MultiActionButton text="Main Button" subtext="Subtext" size="large">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByText("Subtext")).toBeVisible();
+});
+
+test("should render with provided 'data-' tags", () => {
+  render(
+    <MultiActionButton text="Main Button" data-element="foo" data-role="bar">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByTestId("bar")).toHaveAttribute("data-element", "foo");
+});
+
+test("should render with provided 'id'", () => {
+  render(
+    <MultiActionButton text="Main Button" id="test">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toHaveAttribute(
+    "id",
+    "test"
+  );
+});
+
+test("should call 'onClick' when the main button is clicked", async () => {
+  const onClick = jest.fn();
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button" onClick={onClick}>
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+
+  expect(onClick).toHaveBeenCalledTimes(1);
+});
+
+test("should open additional buttons when the main button is clicked", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+
+  expect(await screen.findByRole("button", { name: "First" })).toBeVisible();
+});
+
+test("should close additional buttons when a child button is clicked", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+  await user.click(screen.getByRole("button", { name: "First" }));
+
+  expect(
+    screen.queryByRole("button", { name: "First" })
+  ).not.toBeInTheDocument();
+});
+
+test("should close additional buttons when a click occurs outside the component", async () => {
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
+  await user.click(document.body);
+
+  expect(
+    screen.queryByRole("button", { name: "First" })
+  ).not.toBeInTheDocument();
+});
+
+test("should render when children are non-Carbon Button children", async () => {
```

Comment:
**suggestion**: this could be moved to the top of the component to be grouped with our initial render tests